tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs

```diff
@@ -320,5 +314,45 @@ ImmutableArray<MetadataReference> metadataReferences
 			return solution.Projects.First()
 				.GetCompilationAsync().Result;
 		}
+
+		private sealed record PrettyDiagnostic(
+			string Id,
+			LinePositionSpan LinePosition,
+			string Message
+		) {
+
+			/// <remarks>
+			/// Formatting output for Visual Studio Test Explorer
+			/// </remarks>
+			public override string ToString() {
+
+				StringBuilder sb = new StringBuilder();
+				const string indent = "\t\t";
+
+				void WriteProperty( string name, object value, string seperator = "," ) {
+					sb.Append( indent );
+					sb.Append( name );
+					sb.Append( " = " );
+					sb.Append( value );
+					sb.AppendLine( seperator );
+				}
+
+				sb.AppendLine( "{" );
+				WriteProperty( nameof( Id ), Id );
+				WriteProperty( nameof( LinePosition ), LinePosition );
+				WriteProperty( nameof( Message ), Message, seperator: string.Empty );
+				sb.AppendLine( "}" );

```

Comment:
I could imagine adding the actual sourceline with an arrow pointing to the position in the future, so, cool setup.